tracee/tracee.go

```diff
@@ -285,6 +333,31 @@ func (t *Tracee) initBPF(ebpfProgram string) error {
 				}
 				continue
 			}
+			if probe.attach == rawTracepoint {
+				// We fallback to regular tracepoint in case kernel doesn't support raw tracepoints (< 4.17)
```

Comment:
For kernels > 4.17, yes. Otherwise (4.14-4.17) we fallback to "regular" tracepoints

Reply:
I think that the fallback logic is specific to out usage of sysenter and sysexit, and it depends on the correct functions being declared in C. if we (or others) add raw traceopints in the future they have to know about this not fallback logic and adjust their code for it. isn't it better to leave this code simple (if kernel doesn't support it, raise error like for other cases), and move the fallback logic earlier, so that this code receives a correct and relevant probe to attach.

Reply:
In general, you are right, but there is a problem in doing that - raw tracepoints don't have category. This means that, for example, to define sys_enter/sys_exit as a raw tracepoint, the category name "raw_syscalls" isn't required.
In order to make things work without changing the events maps, I had to mix between the two and do the split here.
My thought was, that we will always want to implement tracepoints as a raw tracepoint, and fallback to regular tracepoint when kernel version doesn't support this.
In the future, we can remove the regular tracepoints support (when tracee will support kernel >4.17)

Reply:
you are already able to handle the missing category etc here, so the same code could be used, just moved elsewhere. then this attachment code would receive the probe object with the relevant attachment info for the current system just as a normal event in eventstotrace.

Reply:
Ok, I'll try to do that in my next commit